### PR TITLE
Chore: bump golangcilint to 1.43 and apply updated code suggestions

### PR DESCRIPTION
**Pull request by dduportal**
This PR comes from the work in #412 .

It updates the golantci-lint configuration file to support the latest 1.43.x line.

It also applies some of its suggestions.

## Test

To test this pull request, you can run the following commands:

```shell
go build -o dist/updatecli
./dist/updatecli diff --config ./examples/updatecli.generic/file.yaml
./dist/updatecli diff --config ./examples/updatecli.generic/yaml.yaml
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -121,6 +122,6 @@ run:
 # golangci.com configuration
 # https://github.com/golangci/golangci/wiki/Configuration
 service:
-  golangci-lint-version: 1.27.x   # Use the fixed version to not introduce new linters unexpectedly
+  golangci-lint-version: 1.43.x   # Use the fixed version to not introduce new linters unexpectedly
```
Comment (olblak) [inline .golangci.yml:125]: Can I ask you to also add an updatecli configuration to bump this version.
It should be pretty straightforward https://github.com/golangci/golangci-lint/releases/tag/v1.43.0

**Review by olblak — DISMISSED** [review]
Thanks for looking into that, I have been pushing back this for too long. 
I made a minor suggestion that could help us to keep track of new golangci new versions.
The current PR is good as is

File: .golangci.yml (hunk 1 above)
Comment (dduportal) [reply .golangci.yml:125]: Don't merge: I'll add a commit with this change

Comment (dduportal) [reply .golangci.yml]: PR upgraded: in fact we won't need an automated version.
I've deleted the `service:` section in `.golangci.yml` as per https://github.com/golangci/golangci/wiki/Configuration#service-configuration : we don't need it.
Thus we don't need to specify the golangcitlint version as well.

The only other location where we specify a fixed version for golangcilint is the githubaction but it's tracked by dependandabot.

So it sounds like we should be ready to roll.

**Pull request merged**